### PR TITLE
LPS-110925 Update modal usages in fragment-web

### PR DESCRIPTION
**Pull request by markocikos**
Hey @wincent ,

In this PR, we are applying `openModal` utility to `fragment-web`, extending API when needed. As a side-task, we are changing babel inclusion configuration, so we can have `.scss` imports and `.js files`.

This is a followup on:
1. https://github.com/wincent/liferay-portal/pull/192
1. https://github.com/jbalsas/liferay-portal/pull/2154
1. https://github.com/markocikos/liferay-portal/pull/182

I'll extend original test instructions, as there are two modals affected:
1. Go to Control Panel > Site > Site Builder > Page Fragments
1. In dropdown menu next to 'Collections +', choose 'Import' <-- first modal
1. Create a new collection, named 'My Collection'
1. In dropdown menu next to 'My Collection', choose 'Import' <-- second modal

Notes:
- We are not updating all modals in `fragment-web`, there are `SimpleInputModal` implementations, but those can be a separate task.
- The styles additions are a fix when AUI modal and `ClayModal` are on the same page, which is the case on the Page Fragments page.
- We are exposing `openModal` globally as `Liferay.Util.openModal`. It fails with normal import, with `Can't resolve 'frontend-js-react-web'` error. So, I used the same import magic as  `Liferay.Util.openToast`, without delving too deep to understand what's going on. I'm not sure if it is possible to make normal import work, but it would be a welcome improvement.
- We are using `onClose` in API, replacing `dialog.after.destroy`. `destroy` is a dialog lifecycle state in AUI, but I feel `onClose` far better matches the intended use than something like `afterComponentUnmount`, or leaving it as it is.
- There was a problem with `cancel` and `import` buttons, in the modal footer. They were a part of iframe. We need `cancel` button to interact with React component, closing it, and `import` button to submit form within iframe. If they are in iframe, submit button works fine but what to do with cancel? I couldn't find how AUI or our portal does this, but I'm guessing it is some global listener that closes modal. As I didn't like that path, I took an alternative, to move the buttons from iframe to `ClayFooter`. I feel that this placement better matches their concern. It slightly changed styles, standardizing paddings. It made `cancel` button relatively easy to implement, but now `submit` needed to be more complicated, as we need to find the form within iframe. I'm pretty sure in almost all cases there is only one form withing iframe, we can modify code if we ever find an exception.
- We are using `buttons` in API, replacing `toolbars.footer`. This is used 15 times in portal, with the sole purpose of defining buttons. Other API of `toolbars` is not used. Not much to say here, renaming makes sense to me.
- We are using the `cancel` button type. If a button with button type `cancel` is defined on modal, it closes the modal. This is, of course, not a standard button type, but a part of AUI button API. It is a very common use case, 226 uses in portal. I like it.
- The rest of the `button` API matches `ClayButton` API. The only difference is the special `cancel` type. I added `id` only for the purposes of a test, but it will be needed down the line in usages.

Please review the code.

Thank you!

/cc @jbalsas

### CODE REVIEW

**Comment by wincent** [comment]
> * We are exposing `openModal` globally as `Liferay.Util.openModal`. It fails with normal import, with `Can't resolve 'frontend-js-react-web'` error. So, I used the same import magic as  `Liferay.Util.openToast`, without delving too deep to understand what's going on. I'm not sure if it is possible to make normal import work, but it would be a welcome improvement.

I suspect it is the same root cause as @jbalsas dealt with [here](https://github.com/wincent/liferay-portal/pull/198/commits/4846ba80822322dc2b9ad5426fe0d10ccc24a935). If the file doesn't end in `*.es.js` it doesn't get included. As far as I can tell, there are only two modules in liferay-portal that follow this pattern (of excluding everything and then un-blacklisting `*.es.js`). I think we should probably head in the direction we want to go (ie. of naming using `*.js`) and deal with the hassle of having to do some more un-blacklisting in this module, until we get to the point where all the AUI is gone and we can remove the whole `ignore` config.

**Comment by wincent** [comment]
> * As I didn't like that path, I took an alternative, to move the buttons from iframe to `ClayFooter`. I feel that this placement better matches their concern.

This sounds like the right call to me. 👍

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js [hunk 1]
```diff
@@ -49,20 +52,96 @@ const Modal = ({id, size, title, url}) => {
...
+		if (type === 'cancel') {
+			processClose();
+		}
+		else if (url && type === 'submit') {
+			const iframe = document.querySelector('.liferay-modal iframe');
+
+			if (iframe) {
+				const form = iframe.contentWindow.document.querySelector(
+					'form'
+				);
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js:65]: Any thoughts on what might happen if somebody sticks more than one `form` in the modal *and* configures a submit button? (Seems unlikely, but I was wondering if we should instead use `querySelectorAll` and if the found count is `!== 1`, log a warning when `process.env.NODE_ENV === 'development'`.)

**Comment by wincent** [comment]
> > * We are exposing `openModal` globally as `Liferay.Util.openModal`. It fails with normal import, with `Can't resolve 'frontend-js-react-web'` error. So, I used the same import magic as  `Liferay.Util.openToast`, without delving too deep to understand what's going on. I'm not sure if it is possible to make normal import work, but it would be a welcome improvement.
> 
> I suspect it is the same root cause as @jbalsas dealt with [here](https://github.com/wincent/liferay-portal/pull/198/commits/4846ba80822322dc2b9ad5426fe0d10ccc24a935). If the file doesn't end in `*.es.js` it doesn't get included. As far as I can tell, there are only two modules in liferay-portal that follow this pattern (of excluding everything and then un-blacklisting `*.es.js`). I think we should probably head in the direction we want to go (ie. of naming using `*.js`) and deal with the hassle of having to do some more un-blacklisting in this module, until we get to the point where all the AUI is gone and we can remove the whole `ignore` config.

Aaaaaand I see you already figured this out in https://github.com/wincent/liferay-portal/pull/198/commits/d2cfc0b3120502a8bb936311ffd459742778ec8b 😂

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js:65]: Normally, this should not happen, there was always a strong convention of having only a single form on one page. But, I wouldn't rule out that case does exist somewhere. If we do stumble upon a case, my idea would be to extend button API with `formId` and query that. A warning in any case is a good idea!

**Comment by wincent** [comment]
Sadly I can't test this @markocikos today. I have two separate portal repos and both are refusing to load any pages for me today with `2020-04-06 12:51:08.399 ERROR [http-nio-8081-exec-2][status_jsp:844] A servlet matching the filter (servlet.init.private=false)(servlet.type=friendly-url) is unavailable`. Tried both master (twice) and your PR, with `git clean` and `ant all` thrown in for good measure. 😭

**Comment by wincent** [comment]
@markocikos: I may be using this wrong, but I don't think it is working as intended. After importing, the modal doesn't close, and I can create nested modals:

1. Import a collection:

00-first-modal

2. After importing, modal is still open, but with page content inside it:

01-after-import-nested

3. If I import again, I get a modal inside a modal:

02-import-again-nested-modal

4. Same with a custom collection; I can import:

03-second-modal

5. But modal doesn't close:

04-after-import

6. And I can create a modal in a modal if I import again:

05-import-again-nested-modal

7. I can go as deep as I want...

06-deep-recursion

**Comment by markocikos** [comment]
@wincent I cannot reproduce this issue. Your step 2 is where something goes wrong. The correct behavior is to have success message in the modal, while modal content is still the choose file form:

screen

After that, your only option is to close the window, followed by page refresh after close.

**Comment by markocikos** [comment]
See https://github.com/wincent/liferay-portal/pull/203

**Pull request closed without merging**